### PR TITLE
Feature intl + reverse proxy

### PR DESCRIPTION
**Pull request by floviolleau**

### CODE REVIEW

**Comment by AndreMiras** [comment]
Hi @floviolleau, thanks a mil for the PR.
It's a bit too big and I may have too many comments.
Here are a few questions/comments I have already:
- make a PR with only the reverse proxy part (I'm currently not clear what's doing)
- one with only the translation
- why the dependency reorg?
- for the linting we already use `prettier`, use `yarn format` to apply
- not clear to me why is the fireplace regex change

File: src/utils/hooks.ts [hunk 1]
```diff
@@ -13,8 +13,7 @@ const useIsLoggedIn = (): boolean | undefined => {
   const { token } = useContext(TokenContext);
   // application is still loading
   if (token === undefined) return undefined;
-  const tokenValid = typeof token === "string" ? token.length > 0 : false;
-  return tokenValid;
+  return typeof token === "string" ? token.length > 0 : false;
```
Comment (AndreMiras) [inline src/utils/hooks.ts:16]: I purposely introduced that `tokenValid` variable to give it some semantic

File: src/utils/helpers.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const fireplaceRegex = /[0-9A-Fa-f]{12}/g;
+const fireplaceRegex = /[0-9a-f]{12}/g;
```
Comment (AndreMiras) [inline src/utils/helpers.ts:1]: I see the commit says `feat: change a bit the regex on mac address for correct error message`, but it's not clear to me why the message is correct if we disallow capital case in the fireplace hex ID?

File: package.json [hunk 3]
```diff
@@ -1,37 +1,50 @@
 {
   "name": "edilkamin",
   "homepage": "https://andremiras.github.io/edilkamin/",
-  "version": "2022.10.20",
+  "version": "2023.02.01",
```
Comment (AndreMiras) [inline package.json:4]: Version bump is done after during the release process

File: next.config.js [hunk 4]
```diff
@@ -1,7 +1,18 @@
+const {i18n} = require('./next-i18next.config')
+
+const tsconfigPath = process.env.NEXTJS_TSCONFIG_PATH
+    ? process.env.NEXTJS_TSCONFIG_PATH
+    : './tsconfig.json'
```
Comment (AndreMiras) [inline next.config.js:5]: YAGNI, I'd rather keep PR small and introduce change incrementally as we need them

**Review by AndreMiras — COMMENTED** [review]
Added some specific comments to give an idea as the PR gets broken down, but that deserves to be break down into different PR.
I understand that's some work so if you don't feel like it I can try to rework it myself by cherry picking and addressing the comments if I ever get time for it.
Also the formatting change makes it harder to review, we already use `prettier` via `yarn format` as formater

**Comment by floviolleau** [comment]
Hi,

Just a quick answer. I'm a bit busy right now. I will take care of all remarks soon

Regards

File: src/utils/helpers.ts (hunk 2 above)
Comment (floviolleau) [reply src/utils/helpers.ts:1]: On my side I got only lowercase mac address so with the previous regex, it wasn't working for me.

File: src/utils/hooks.ts (hunk 1 above)
Comment (floviolleau) [reply src/utils/hooks.ts:16]: We can a comment instead.

I did it because it is useless to have a variable just returned after.

**Comment by floviolleau** [comment]
Hi,

I'm not forgetting this PR 😉

I'm very busy at work and I don't really want to spend time to split the PR into more PR because I did lot's of manual tests.

See this PR as architecture changes to put foundations for future devs 😉.

I will take care of your comments of the review.

Why adding proxy support? It's because I have a server that proxy all requests to corresponding server and doing the dispatch. 

I did a reorg of dependencies because some where in deps instead of being in devDeps and as far as I remember, I organized it alphabetically (and added more)

I didn't checked prettier.

Kind regards

**Comment by AndreMiras** [comment]
Thanks for getting back.
I understand you're busy on other things and thanks again for taking time to look into it.
Then if you don't mind I can cherry pick your commits later then rework and integrate them as I see fit, you would still appear as the author of the commit I would cherry pick

**Comment by AndreMiras** [comment]
I've worked on the project recently as I have a bit of time.
If you're still interested and have time, would you mind to make a MVP with minimal code change touching only what's needed so we can review/merge it more easily? The MVP could make only a single string translation as a demo and then in next iteration we could bring up other component changes.
Otherwise no worries I can try to work out through your changes and cherry-pick/rebase them

**Comment by floviolleau** [comment]
Hi Andre,

I think it is too much work to split everything because all is linked and I don't think it worth it to make it.
So please kindly approve the PR 😊

I made new changes like migrating next JS to latest as well on my side. I had a lot of issues doing it and I spent already a lot of time on it.

What I didn't pushed yet but I already did is a doing a proper login page and I started doing a back-end admin interface with react admin https://marmelab.com/react-admin

I think we need to communicate better because we are doing twice the work.

WDYT?

Have a good day

**Comment by floviolleau** [comment]
I didn't have chances to write translations yet because I'm focused on features.

Here some captures of the WIP:

image

image
image

**Comment by AndreMiras** [comment]
Hey Florent,
Yes I understand this is too much work indeed.
If you don't mind I'll cherry-pick and co-author you later then for the translation part.

Yes it's a shame to do the same twice indeed, even though that's what happen in open source sometimes.
I get this can be frustrating. At least we're open sourcing. We can still borrow each other work meanwhile taking different directions.

Your new login page looks awesome anyway, keep it up with the good work :fire: 
What would the backend admin used for by the way?

**Comment by floviolleau** [comment]
Hi,

Yes. I share what you wrote.

My time is limited for opensource and I think yours as well so if we can move in the same direction, I will be glad to help 😊.

I will update this PR with your latest changes but I'm facing to insane merge conflicts and I do not want to not loose my job and having them integrated in the main.

I suggest, if you agree, to merge this PR (once update) so we can share the same code base for the future because it is part of the foundations (especially translations).

Back-end admin is just the same interface you did but behind a login page and organize a bit better all information (if you agree).

Can we do that before you do heavy changes in code base?

**Comment by AndreMiras** [comment]
Thanks for trying to solve the huge amount of conflicts.
Leave it there I'll cherry pick the things that I think make sense for now.
I will try to make a translation MVP cherry-picking your changes, but I don't want all the other changes regarding code styling or other package update.
I may get there eventually, but I won't address that at once with a big PR merge.

**Comment by floviolleau** [comment]
Okey, so I give up.

You will see that adding translations introduce big changes everywhere. This is mostly why this PR is pretty big.

Moreover, I autoindent by using Webstorm and I'm using usual code convention regarding indentation and spaces.

Using double quotes is not encouraged, prefer template strings, prefer removing extra spacings around curly brackets for named imports, etc.

Have a good day

**Pull request closed without merging**